plugins/@grouparoo/calculated-property/src/lib/profileProperty.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  plugin,
+  Profile,
+  ProfileProperty,
+  ProfilePropertyPluginMethod,
+} from "@grouparoo/core";
+import { NodeVM } from "vm2";
+export interface RequiredPropertiesObject {
+  [propertyId: string]: ProfileProperty;
+}
+
+async function calculateProfilePropertyValue(
+  customFunction,
+  profile: Profile
+): Promise<string> {
+  let populatedFunction;
+  try {
+    populatedFunction = await plugin.replaceTemplateProfileVariables(
+      customFunction,
+      profile,
+      false
+    );
+    //returns a string of the entire function
+  } catch (error) {
+    //if we don't have the right properties to build the function, bail
+    console.log(error);
+    return undefined;
+  }
+  let calculatedPropertyValue;
+
+  //this works with vm and node vm... using node vm to allow for node modules in v2 of this feature
+  const vm = new NodeVM({
+    console: "inherit",
+    sandbox: {},
+    argv: [],
+    env: {},
+  });
```

Comment:
We're allowing console usage, but keeping everything else locked down.  External libraries cannot be imported and neither can env variables.